include/net/dsa.h

```diff
@@ -0,0 +1,174 @@
+/** @file
+ @brief DSA definitions and handlers
+
+ */
+
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *               Lukasz Majewski <lukma@denx.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_DSA_H_
+#define ZEPHYR_INCLUDE_NET_DSA_H_
+
+#include <device.h>
+#include <net/net_if.h>
+
+/**
+ * @brief LLDP definitions and helpers
+ * @defgroup lldp Link Layer Discovery Protocol definitions and helpers
+ * @ingroup networking
+ * @{
+ */
+
+#define NET_DSA_PORT_MAX_COUNT 8
+
+#define DSA_NET_DEVICE_INIT(dev_name, drv_name, init_fn, pm_control_fn,	\
+			    data, cfg_info, prio, api, mtu, port_num)	  \
+	DEVICE_DEFINE(dev_name, drv_name, init_fn, pm_control_fn, data,	\
+		      cfg_info, POST_KERNEL, prio, api);		\
+	NET_L2_DATA_INIT(dev_name, 0, NET_L2_GET_CTX_TYPE(ETHERNET_L2)); \
+	NET_IF_INIT(dev_name, 0, ETHERNET_L2, mtu, port_num)
+
+#define DSA_STATUS_TICK_MS K_MSEC(1000)
```

Comment:
The `TICK` here is confusing as the timeout API will use ticks internally, but here we have hybrid ticks ms. Not sure where this is used but could we figure out better name for it.

Reply:
This time is used with k_delayed_work_submit(), to check if the ethernet cable was removed/plugged to DSA slave port. I'm going to rename it to DSA_STATUS_PERIOD_MS.